client/rest-high-level/src/main/java/org/elasticsearch/client/slm/SnapshotLifecyclePolicyMetadata.java

```diff
@@ -77,6 +80,9 @@
         PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), SnapshotInvocationRecord::parse, LAST_FAILURE);
         PARSER.declareLong(ConstructingObjectParser.constructorArg(), NEXT_EXECUTION_MILLIS);
         PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), SnapshotInProgress::parse, SNAPSHOT_IN_PROGRESS);
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(),
+            (p, c) -> SnapshotLifecycleStats.SnapshotPolicyStats.parse(p, "policy"), POLICY_STATS);
```

Comment:
So the stats requires a policy name, since it's a "`NamedObject`" from the parser's perspective, but the parser doesn't like parsing it this way.

For now, it's hardcoded as the policy ID (since in this case there is only ever one policy stats it can be safely ignored. In the future though, we should look at changing the format to be an array of objects rather than using dynamic keys. I'll add this as a TODO to the meta issue.